src/osd/OSD.cc

```diff
@@ -9184,7 +9184,7 @@ void OSD::do_notifies(
     dout(7) << __func__ << " osd." << it->first
 	    << " on " << it->second.size() << " PGs" << dendl;
     MOSDPGNotify *m = new MOSDPGNotify(curmap->get_epoch(),
-				       it->second);
+				       std::move(it->second));
```

Comment:
Basically this is about `notify_list` taken by non-const reference. `OSD::do_notifies` can be called only from `OSD::dispatch_context` which is turn has 3 users in `OSD.cc`. In each of them `PG::RecoveryCtx` are freshly created.

Also the two `MOSDPGNotify` instances are brand-new, so their `pg_list` members must be empty. In consequence there shouldn't be a difference between `swap` and `move`.

 Seems to be fine.

Reply:
yeah, the current semantic is also easier to understand. the notify_list is supposed to be consumed by `do_notifies`.